File: src/main/java/com/example/demo/Entity/InvoiceDetailTable.java
package com.example.demo.Entity;



import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "invoice_detail_table")
@Data
public class InvoiceDetailTable {
	
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "invdtl_id")
	private long invdtlId;
	
    @ManyToOne
    @JoinColumn(name = "invoice_id")
    private InvoiceHeaderTable invoice;

    @ManyToOne
    @JoinColumn(name = "addon_id")
    private AddOnMaster addon;

    @Column(name = "addon_amt")
    private double addonAmt;

}

File: Backend.NET/Models/InvoiceDetailTable.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FleetManagementSystem.Api.Models;

[Table("invoice_detail_table")]
public class InvoiceDetailTable
{
    [Key]
    [Column("invdtl_id")]
    public long InvdtlId { get; set; }

    [ForeignKey("Invoice")]
    [Column("invoice_id")]
    public long? InvoiceId { get; set; }
    public InvoiceHeaderTable Invoice { get; set; }

    [ForeignKey("AddOn")]
    [Column("addon_id")]
    public int? AddOnId { get; set; }
    public AddOnMaster AddOn { get; set; }

    [Column("addon_amt")]
    public double AddonAmt { get; set; }
}
